KomodoPlatform/komodo-wallet-desktop
# Add Maker mode, and order form options for `GoodTillCancelled` / `FillOrKill`

Closes https://github.com/KomodoPlatform/komodo-wallet-desktop/issues/956

This PR also includes some changes to the dex pro view layout. All trading operations on both Simple and Pro view should be thoroughly tested for regression.


https://github.com/KomodoPlatform/komodo-wallet-desktop/assets/35845239/f9a436fb-0bc7-4130-bb9a-f43ceaae80ba

Beyond the layout changes, this PR introduces a checkbox in the `Confirm Exchange Details` modal 
![image](https://github.com/KomodoPlatform/komodo-wallet-desktop/assets/35845239/b90fac8c-cbb3-4b77-8844-ab0cef6124df)

This checkbox is on by default, which is effectively the same as before: When you create an order, if it does not match within 30 seconds, it becomes a maker order and stays on the orderbook.

If the checkbox is unticked, the order id place with the `FillOrKill` [order_type parameter.](https://komodo-docs-revamp-2023.pages.dev/en/docs/atomicdex/api/common_structures/orders/#order-type) 

This means that if an order is placed which is not matched with an existing order on the orderbook, it will automatically cancel after 30 seconds.

To Test (pro view):
- Place [orders which will not be matched](https://github.com/KomodoPlatform/komodo-wallet-desktop/pull/2420):
   - in sell mode, with good until cancelled checkbox ticked on. It should remain on the orderbook after trying and failing to match.
   - in sell mode, with good until cancelled checkbox ticked off. It should not enter the orderbook after trying and failing to match.
   - in buy mode, with good until cancelled checkbox ticked on. It should remain on the orderbook after trying and failing to match.
   - in buy mode, with good until cancelled checkbox ticked off. It should not enter the orderbook after trying and failing to match.

To Test (general):
- Attempt some orders in simple view to confirm no regression.
- Navigate around the new pro view layout to confirm all components still function as expected.

PART 2 BELOW
